src/python/dxpy/sugar/xfer.py

```diff
@@ -0,0 +1,806 @@
+    else:
+        return_filename = input_filename[:-endlen]
+    return_filename = SPECIAL_RE.sub("", return_filename)
+    local_filename = "tar_output_{0}".format(return_filename)
+    if output_dir:
+        return_filename = os.path.join(output_dir, return_filename)
+
+    tar_cmd.extend([">", local_filename])
+
+    cmds = [["dx", "download", remote_file.get_id(), "-o", "-"], tar_cmd]
+
+    info("Downloading file {} to {} using command {}".format(
+        remote_file.get_id(), return_filename, cmds
+    ))
+    info("Saving command stdout to intermediate file {}".format(local_filename))
+
+    result = chain(cmds, block=block)
+
+    if block:
+        return return_filename
+    else:
+        return return_filename, result[-1]
+
+
+def download_and_decompress_file(
+    remote_file, input_filename=None, local_filename=None, output_dir=None,
+    project=None, block=True
+):
+    """
+    Download and decompress a gzipped file.
+
+    Args:
+        remote_file (dxpy.DXFile or dxlink): DNAnexus link or file-id of file to
+            download.
+        input_filename (str): Name to use for the input filename, if different than
+            the name of the input_file. If not provided, platform filename is used.
+        local_filename: Local filename/dirname. If not None, this file/directory must
+            exist after unpacking the archive or an error is raised.
+        output_dir (str): Download file to a specific directory (default is the current
+            directory).
+        project: The project that contains the file, if it is not the currently
+            selected project and is not specified in the remote file object/link.
+        block: Wait for the download to complete before returning.
+
+    Returns:
+        The path of the decompressed file. Also returns the running download process if
+        `block is False`.
+    """
+    remote_file = _as_dxfile(remote_file, project)
+
+    if input_filename is None:
+        input_filename = remote_file.describe()["name"]
+
+    if local_filename is None:
+        local_filename = input_filename[:-3]
+    if output_dir:
+        local_filename = os.path.join(output_dir, local_filename)
+
+    cmds = [
+        ["dx", "download", remote_file.get_id(), "-o", "-"],
+        ["gunzip", ">", local_filename]
+    ]
+
+    info(
+        "Downloading file {} to {} using command {}",
+        remote_file.get_id(), local_filename, cmds
+    )
+
+    result = chain(cmds, block=block)
+
+    if block:
+        return local_filename
+    else:
+        return local_filename, result[-1]
+
+
+def _as_dxfile(fileobj, project=None):
+    if dxpy.is_dxlink(fileobj):
+        project_id = None
+        if (
+            isinstance(fileobj["$dnanexus_link"], dict) and
+            "project" in fileobj["$dnanexus_link"]
+        ):
+            project_id = fileobj["$dnanexus_link"]["project"]
+        elif project:
+            project_id = project.get_id()
```

Comment:
Doesn't seem to me like consumers of this function are supplying a `dxpy.DXProject`. It looks like they're supplying a `str` project-id directly.

The documentation for those funtions makes it seem like a `str` is expected.